### ISSUE TITLE
Version 1.3.0 failing on compile_time

### ISSUE THREAD

**Issue by rmldsky** [chef-cookbooks/xml#29]
I do not use `xml` cookbook directly, it is fetched by `route53` cookbook as dependency. On the same machine when I use version `1.2.13` all is OK. After cookbook upgrade to `1.3.0`, `chef-client` run fails with below error (chef-client is 11.18.12, system is Ubuntu 14.04):
```
Recipe: xml::default
  * package[libxml2-dev] action install (up to date)
  * package[libxslt-dev] action install (up to date)
  * package[zlib1g-dev] action install (up to date)

  ================================================================================
  Recipe Compile Error in /var/chef/cache/cookbooks/XXX/recipes/XXX.rb
  ================================================================================

  ArgumentError
  -------------
  wrong number of arguments (1 for 0)

  Cookbook Trace:
  ---------------
    /var/chef/cache/cookbooks/xml/recipes/ruby.rb:46:in `block in from_file'
    /var/chef/cache/cookbooks/xml/recipes/ruby.rb:43:in `from_file'
    /var/chef/cache/cookbooks/route53/recipes/default.rb:20:in `from_file'
    /var/chef/cache/cookbooks/XXX/recipes/_XXX.rb:1:in `from_file'

  Relevant File Content:
  ----------------------
  /var/chef/cache/cookbooks/xml/recipes/ruby.rb:

   39:      ENV['NOKOGIRI_USE_SYSTEM_LIBRARIES'] = node['xml']['nokogiri']['use_system_libraries'].to_s
   40:    end
   41:  end
   42:
   43:  chef_gem 'nokogiri' do
   44:    version node['xml']['nokogiri']['version']
   45:    action :install
   46>>   compile_time true if respond_to?('compile_time')
   47:  end
   48:


  Running handlers:
[2015-10-12T10:22:11+00:00] ERROR: Running exception handlers
  Running handlers complete
[2015-10-12T10:22:11+00:00] ERROR: Exception handlers complete
[2015-10-12T10:22:11+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
  Chef Client failed. 1 resources updated in 20.492634483 seconds
[2015-10-12T10:22:11+00:00] ERROR: wrong number of arguments (1 for 0)[2015-10-12T10:22:11+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

**Comment by tas50** [comment]
Sorry about that.  It looks like due to the compile_time true change this is now a Chef 12+ cookbook.  I've updated the readme to reflect this and I'll release a new version with a note in the Changelog.  For Chef 11 you'll need to stay on the 1.2 release, but I would highly suggest updating to Chef 12 since there's been a huge number of great changes since Chef 11.

**Comment by tas50** [comment]
I spoke too soon.  I fixed the Chef 11 compat with 1.3.1.  I still highly recommend you upgrade to Chef 12.  In the near future we will require Chef 12 for all community cookbooks.

**Comment by rmldsky** [comment]
Thanks for putting your time into this issue! Upgrading to Chef 12+ is already a work in progress!

Cheers and have a good one!
